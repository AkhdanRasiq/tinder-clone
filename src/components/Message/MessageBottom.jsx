import React from 'react'

import TextField from '@mui/material/TextField'
import SendIcon from '@mui/icons-material/Send'
import { IconButton } from "@mui/material"

import { onSendWebsocket } from '../../utils/adapters'
import { handleGenerateMessage } from '../../utils/util'

class MessageBottom extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      strMessage: "",
      arrMessages: []
    }
  }

  componentDidUpdate = (prevProps) => {
    if(this.props.a_arrMessages !== prevProps.a_arrMessages)
      this.setState({ arrMessages: this.props.a_arrMessages })
  }

  handleMessageToSend = () => {
    const generatedMessage = handleGenerateMessage(this.state.strMessage)
    onSendWebsocket(generatedMessage);

    this.setState({ arrMessages: [...this.state.arrMessages, generatedMessage]})
    this.props.a_messageOnUpdateCallback([...this.state.arrMessages, generatedMessage])

    this.setState({ strMessage: "" })
  }

  handleSendMessage = (event) => {
    if (event.key === 'Enter' && this.state.strMessage !== '')
      this.handleMessageToSend()
  }

  handleBtnSendMessage = (event) => {
    event.preventDefault()
    if (this.state.strMessage !== '')
      this.handleMessageToSend()
  }

  handleTyping = (event) => {
    this.setState({ strMessage: event.target.value });
  }

  render() {
    return (
      <div className    ='messageBottomContainer'>

        <TextField
          fullWidth
          className     ='messageBottom_textfield'
          placeholder   ='Type a message'
          autoComplete  ='off'
          onKeyDown     ={this.handleSendMessage} 
          onChange      ={this.handleTyping}
          value         ={this.state.strMessage}  
        />

        <IconButton onClick={this.handleBtnSendMessage}>
          <SendIcon
            className   ='messageBottom_btnSend'
            fontSize    ='large'
          />
        </IconButton>
      </div>
    )
  }
}

export default MessageBottom
