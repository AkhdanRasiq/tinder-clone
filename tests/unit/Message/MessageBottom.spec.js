import React from 'react'
import { shallow } from "enzyme"

import MessageBottom from "../../../src/components/Message/MessageBottom"

const test_a_messages = {
  text      : "Test",
  user      : "Admin",
  timestamp : 1643468925
}

describe("MessageBottom.jsx", () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <MessageBottom
        a_arrMessages={[{
            ...test_a_messages
        }]}
      />
    )
  })

  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("All Components is Exist", () => {
    const tf_message  = wrapper.find('.messageBottom_textfield')
    const btn_message = wrapper.find('.messageBottom_btnSend')

    expect(tf_message.length).toBe(1)
    expect(btn_message.length).toBe(1)
  })

  it("Typing Message", () => {
    const tf_message  = wrapper.find('.messageBottom_textfield').at(0)

    tf_message.simulate('change', { target: { value: 'Test test test' } })
    expect(wrapper.state('strMessage')).toEqual('Test test test')
  })
})
