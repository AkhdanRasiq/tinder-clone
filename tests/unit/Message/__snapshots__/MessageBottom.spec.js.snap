// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBottom.jsx renders properly 1`] = `ShallowWrapper {}`;
